de.fu_berlin.inf.dpp.server/src/de/fu_berlin/inf/dpp/server/SarosServer.java

```diff
@@ -22,19 +22,16 @@ public SarosServer() {
     lifecycle = new ServerLifecycle();
   }
 
-  public void start() {
+  public void start(ServerConsole console) {
 
     // Logging
     URL log4jProperties = SarosServer.class.getResource(LOGGING_CONFIG_FILE);
     PropertyConfigurator.configure(log4jProperties);
 
+    lifecycle.setConsole(console);
```

Comment:
I guess the context has a console but not the lifecycle.

Reply:
I first suggest you change the CTOR to InputStream and OutputStream, you can wrap the outputstream inside the ServerConsole.

Next you just do this in the contextfactory: component.add(new ServerConsole(System.in, System.out)


The question is on how you would to register commands. You could also throw each command into the context:
public Mycommand(ServerConsole console) {console.registerCommand(this);}

or you add them through the lifecycle. The choice is up to you.

After looking at the code I just wonder why you have an identifier() method along with a matches() method. And I highly suggest that you throw away the command and only passes the arguments into the command (either already splitted or as string).

Reply:
The console can then be obtained through: server.lifeCycle().getContext().getComponent(ServerConsole.class)

Reply:
> After looking at the code I just wonder why you have an identifier() method along with a matches() method.

The idea was to allow classes to handle multiple related commands to avoid code duplication. Looking at the current implementations, that seems over-engineered.